modules/siddhi-core/src/main/java/io/siddhi/core/table/record/AbstractQueryableRecordTable.java

```diff
@@ -565,94 +515,50 @@ public CompiledCondition compileCondition(Expression condition, MatchingMetaInfo
-                            variableExpressionExecutors));
+                    ((CacheTable) cacheTable).generateCacheCompileCondition(condition, matchingMetaInfoHolder,
+                            siddhiQueryContext, variableExpressionExecutors));
             return new RecordStoreCompiledCondition(expressionExecutorMap, compiledConditionWithCache);
         } else {
             return new RecordStoreCompiledCondition(expressionExecutorMap, compileCondition);
         }
     }
 
     @Override
-    public StreamEvent find(CompiledCondition compiledCondition, StateEvent matchingEvent)
+    public StreamEvent query(StateEvent matchingEvent, CompiledCondition compiledCondition,
+                             CompiledSelection compiledSelection, Attribute[] outputAttributes)
             throws ConnectionUnavailableException {
-        RecordStoreCompiledCondition recordStoreCompiledCondition;
-        CompiledConditionWithCache compiledConditionAggregation = null;
-        if (isCacheEnabled) {
-            RecordStoreCompiledCondition compiledConditionTemp = (RecordStoreCompiledCondition) compiledCondition;
-            compiledConditionAggregation = (CompiledConditionWithCache)
-                    compiledConditionTemp.compiledCondition;
-            recordStoreCompiledCondition = new RecordStoreCompiledCondition(compiledConditionTemp.
-                    variableExpressionExecutorMap, compiledConditionAggregation.getStoreCompileCondition());
-        } else {
-            recordStoreCompiledCondition =
-                    ((RecordStoreCompiledCondition) compiledCondition);
-        }
-
-        Map<String, Object> findConditionParameterMap = new HashMap<>();
-        for (Map.Entry<String, ExpressionExecutor> entry : recordStoreCompiledCondition.variableExpressionExecutorMap
-                .entrySet()) {
-            findConditionParameterMap.put(entry.getKey(), entry.getValue().execute(matchingEvent));
-        }
-
-        Iterator<Object[]> records;
-        if (isCacheEnabled) {
-            assert compiledConditionAggregation != null;
-            readWriteLock.readLock().lock();
-            try {
-                StreamEvent cacheResults = cachedTable.find(compiledConditionAggregation.getCacheCompileCondition(),
-                        matchingEvent);
-                if (cacheResults != null) {
-                    return cacheResults;
+        findMatchingEvent = matchingEvent;
+        if (cacheEnabled && !queryFromStore.get()) {
+            // check if we need to check the size of store
+            if (storeTableSize == -1 || (!cacheExpiryEnabled && storeSizeLastCheckedTime < siddhiAppContext.
+                    getTimestampGenerator().currentTime() - storeSizeCheckInterval)) {
+                StateEvent stateEventForCaching = new StateEvent(1, 0);
+                queryFromStore.set(Boolean.TRUE);
+                try {
```

Comment:
Shall we move this to a different method as its repeatedly used?

Reply:
Won't this queryFromStore.set(TRUE ) overrides the original queryFromStore value that's used in other places of the query?
Here we have to get the original value to a variable and in the final set the original value back to the thread-local. 
Need to fix this in all the preloading places.

Reply:
This is withing an if that checks if queryFromStore is false. so when this is hit queryFromStore is false

Reply:
done